SCons/Environment.py

```diff
@@ -1034,6 +1034,8 @@ def __init__(
         # should override any values set by the tools.
         for key, val in save.items():
             self._dict[key] = val
+        if 'SOURCE_DATE_EPOCH' in os.environ:
+            self._dict['ENV']['SOURCE_DATE_EPOCH'] = os.environ['SOURCE_DATE_EPOCH']
```

Comment:
Not sure that this is going to do what you expect.  This will set a value in the construction environment, yes, but it still won't be passed on to `gcc` or whoever needs to be listening during builds. For that, it needs to be added to the Execution Environment (usually `env['ENV']`).

Reply:
This is what worked in my tests (after [8 failed tries](https://build.opensuse.org/package/revisions/home:bmwiedemann:reproducible:test/scons)). I think, since this is `Environment.py` the `self._dict['ENV']` here **is** `env['ENV']`